File: android/app/src/main/kotlin/com/wakeupsunshine/data/AlarmSound.kt
package com.wakeupsunshine.data

import androidx.compose.ui.graphics.Color
import com.wakeupsunshine.ui.theme.*

/**
 * 4 custom sounds + classic (system fallback). IDs match iOS and backend exactly.
 */
enum class AlarmSound(
    val id: String,
    val displayName: String,
    val description: String,
    val iconName: String,
    val iconColor: Color
) {
    WAKE_UP(
        id = "wake_up",
        displayName = "Wake Up",
        description = "Rise and shine!",
        iconName = "wb_sunny",
        iconColor = Orange500
    ),
    HEY_YOU(
        id = "hey_you",
        displayName = "Hey You Wake",
        description = "Hey, get up!",
        iconName = "music_note",
        iconColor = Green500
    ),
    MINIONS(
        id = "minions",
        displayName = "Minions",
        description = "Banana time!",
        iconName = "sentiment_very_satisfied",
        iconColor = Yellow500
    ),
    RINGING_TONE(
        id = "ringing_tone",
        displayName = "Ringing Tone",
        description = "Classic phone ring",
        iconName = "phone",
        iconColor = Blue500
    ),
    CLASSIC(
        id = "classic",
        displayName = "Classic Alarm",
        description = "Traditional alarm",
        iconName = "alarm",
        iconColor = Red500
    );

    companion object {
        fun from(id: String?): AlarmSound = entries.find { it.id == id } ?: CLASSIC

        val allSounds: List<AlarmSound> get() = entries
        val default: AlarmSound get() = CLASSIC
        val allowedIds: Set<String> get() = entries.map { it.id }.toSet()
    }
}

File: ios/WakeUpSunshine/Models/AlarmSound.swift
import SwiftUI
import AVFoundation

// MARK: - Alarm Sound Model
/// 4 custom sounds + classic (system fallback). IDs must match Android and backend.
enum AlarmSound: String, CaseIterable, Identifiable, Codable {
    case wakeUp      = "wake_up"
    case heyYou      = "hey_you"
    case minions     = "minions"
    case ringingTone = "ringing_tone"
    case classic     = "classic"   // plays system alarm — no file needed

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .wakeUp:      return "Wake Up"
        case .heyYou:      return "Hey You Wake"
        case .minions:     return "Minions"
        case .ringingTone: return "Ringing Tone"
        case .classic:     return "Classic Alarm"
        }
    }

    var description: String {
        switch self {
        case .wakeUp:      return "Rise and shine!"
        case .heyYou:      return "Hey, get up!"
        case .minions:     return "Banana time!"
        case .ringingTone: return "Classic phone ring"
        case .classic:     return "Traditional alarm"
        }
    }

    var iconName: String {
        switch self {
        case .wakeUp:      return "sun.max.fill"
        case .heyYou:      return "hand.wave.fill"
        case .minions:     return "face.smiling.fill"
        case .ringingTone: return "phone.fill"
        case .classic:     return "alarm"
        }
    }

    var iconColor: Color {
        switch self {
        case .wakeUp:      return Color(red: 1.0, green: 0.6, blue: 0.2)
        case .heyYou:      return Color(red: 0.3, green: 0.7, blue: 0.4)
        case .minions:     return Color(red: 0.95, green: 0.8, blue: 0.1)
        case .ringingTone: return Color(red: 0.2, green: 0.5, blue: 0.9)
        case .classic:     return Color(red: 0.9, green: 0.3, blue: 0.3)
        }
    }

    static var `default`: AlarmSound { .classic }
    static var allSounds: [AlarmSound] { allCases }
    static var allowedIds: Set<String> { Set(allCases.map { $0.id }) }

    static func from(id: String?) -> AlarmSound {
        guard let id else { return .default }
        return AlarmSound(rawValue: id) ?? .default
    }
}

// MARK: - WakeRequest Extension
extension WakeRequest {
    var alarmSound: AlarmSound { AlarmSound.from(id: alarmSoundId) }
}
